crates/aiken-project/src/tests/gen_uplc.rs

```diff
@@ -6631,3 +6640,42 @@ fn mk_pair_data() {
         false,
     )
 }
+
+#[test]
+fn pattern_bytearray() {
+    let src = r#"
+        test pattern_bytearray() {
+            let bytes = "foo"
+            when bytes is {
+                "bar" -> False
+                #[0x66, 0x6f, 0x6f] -> True
+                _ -> False
+            }
+        }
+    "#;
+
+    let snd_clause = Term::equals_bytestring()
+        .apply(Term::byte_string(vec![0x66, 0x6f, 0x6f]))
+        .apply(Term::var("__subject"))
+        .delayed_if_then_else(Term::bool(true), Term::bool(false));
+
+    let fst_clause = Term::equals_bytestring()
+        .apply(Term::byte_string(vec![0x62, 0x61, 0x72]))
+        .apply(Term::var("__subject"))
+        .delayed_if_then_else(Term::bool(false), snd_clause);
+
+    let when_clause = fst_clause
+        .lambda("__subject")
+        .apply(Term::var("__when_var"))
+        .lambda("__when_var")
+        .apply(Term::var("bytes"));
+
+    let program = when_clause
+        .lambda("bytes")
+        .apply(Term::byte_string(vec![0x66, 0x6f, 0x6f]))
+        // Not sure what this extra lambda is or do?
+        .lambda("???")
+        .apply(Term::Error.delay());
```

Comment:
@MicroProofs any idea what this extra lambda is? The generatd name don't even seem to be valid UTF-8 sequence or just some plain newline 🤨. It gets eventually eliminated by the optimizer, but I couldn't make sense of what it is.